### PR TITLE
(BOLT-1580) Prune unnecessary spec files from gemfiles in bolt

### PR DESCRIPTION
**Pull request by rileynewton**
Based on current process in pdk-vanagon, https://github.com/puppetlabs/pdk-vanagon/pull/196/commits/3b3eb60eb40459e27d4309f71f6623c99b070eef.

### CODE REVIEW

File: configs/components/gem-prune.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    gem_bins = {
+      settings[:ruby_api] => settings[:host_gem],
+    }
+
+    settings[:additional_rubies]&.each do |rubyver, local_settings|
+      gem_bins[local_settings[:ruby_api]] = local_settings[:host_gem]
+    end
+
+    bolt_ruby_versions = ['2.5.0', '2.7.0']
```
Comment (donoghuc) [inline configs/components/gem-prune.rb:18]: We only ship one ruby version for bolt. You can find that in puppet-runtime in the bolt-runtime project definition https://github.com/puppetlabs/puppet-runtime/blob/341e6f0a196c74d38dda93d0b036b16c4207ca34/configs/projects/bolt-runtime.rb#L4

File: configs/components/gem-prune.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+component 'gem-prune' do |pkg, settings, platform|
+  pkg.build_requires 'bolt-runtime'
+
+  pkg.add_source('file://resources/rubygems-prune')
+
+  pkg.build do
+    puppet_cachedir = File.join(settings[:privatedir], 'puppet', 'ruby')
+    ruby_cachedir = File.join(settings[:cachedir], 'ruby')
```
Comment (rileynewton) [inline configs/components/gem-prune.rb:8]: I'm guessing these are not valid settings in this project.

File: configs/components/gem-prune.rb (hunk 1 above)
Comment (rileynewton) [inline configs/components/gem-prune.rb:18]: I'm guessing the versions that should be here.

File: configs/components/gem-prune.rb (hunk 2 above)
Comment (rileynewton) [reply configs/components/gem-prune.rb]: Adjusted variables to fit `bolt-vanagon`

**Comment by rileynewton** [comment]
This builds and executes the prune successfully, attached a copy of the diff to the ticket https://tickets.puppetlabs.com/secure/attachment/92844/diff-what-was-pruned.txt.

File: configs/components/bolt.rb [hunk 3]
```diff
@@ -3,6 +3,7 @@
 
   # Install the bolt runtime for access to rubygem-r10k:
   pkg.build_requires 'bolt-runtime'
+  pkg.build_requires 'gem-prune'
```
Comment (rileynewton) [inline configs/components/bolt.rb:6]: Not sure this is necessary here or just in puppet-bolt.rb, not sure what the prune process would be for bolt itself versus bolt-vanagon.

Comment (rileynewton) [reply configs/components/bolt.rb:6]: Runs without so looks like not necessary, removing.

**Review by donoghuc — APPROVED** [review]
Poked around in an RPM installed from this. That smoke test looks good. I think its worth it to merge this and see if CI catches anything.

**Pull request merged**